File: android/src/main/kotlin/no/nordicsemi/android/mcumgr_flutter/ext/ImageSlot.kt
package no.nordicsemi.android.mcumgr_flutter.ext

import com.google.protobuf.kotlin.toByteString
import io.runtime.mcumgr.response.McuMgrResponse
import io.runtime.mcumgr.response.img.McuMgrImageStateResponse
import no.nordicsemi.android.mcumgr_flutter.gen.ProtoImageSlot
import okio.ByteString

fun McuMgrImageStateResponse.ImageSlot.toProto(): ProtoImageSlot {
    return ProtoImageSlot(
        slot = this.slot.toLong(),
        version = version,
        hash = ByteString.of(hash),
        bootable = bootable,
        pending = pending,
        confirmed = confirmed,
        active = active,
        permanent = permanent
    )
}

File: lib/models/image_slot.dart
part of mcumgr_flutter;

class ImageSlot {
  final int image;
  final int slot;
  final String? version;
  final Uint8List hash;
  final bool bootable;
  final bool pending;
  final bool confirmed;
  final bool active;
  final bool permanent;

  String get hashString =>
      hash.map((e) => e.toRadixString(16).padLeft(2, '0')).join();

  ImageSlot({
    required this.image,
    required this.slot,
    this.version,
    required this.hash,
    required this.bootable,
    required this.pending,
    required this.confirmed,
    required this.active,
    required this.permanent,
  });
}